src/app/participant-page/participant-page.component.ts

```diff
@@ -1100,20 +1100,22 @@ export class ParticipantPageComponent implements OnInit, OnDestroy {
     );
   }
 
-  getParticipantData(fieldSetting: FieldSettings, relative: ParticipantData) {
-    if (this.participant != null && this.participant.participantData != null && relative.dataId != null && fieldSetting.columnName != null) {
+  getParticipantData(fieldSetting: FieldSettings, personsParticipantData: ParticipantData) {
+    if (this.participant && this.participant.participantData && personsParticipantData && fieldSetting.columnName) {
       let kitFieldsDict = {'DATE_KIT_RECEIVED': 'receiveDate', 'DATE_KIT_SENT': 'scanDate', 'KIT_TYPE_TO_REQUEST': 'kitType'};
       if (fieldSetting.displayType === 'SAMPLE') {
-        let sample: Sample = this.participant.kits.find(kit => kit.bspCollaboratorSampleId === relative.data[fieldSetting.columnName]);
+        let sample: Sample = this.participant.kits.find(kit => kit.bspCollaboratorSampleId === personsParticipantData.data[fieldSetting.columnName]);
         if (sample && kitFieldsDict[fieldSetting.columnName] && sample[kitFieldsDict[fieldSetting.columnName]]) {
           return sample[kitFieldsDict[fieldSetting.columnName]];
         } else {
           return "";
         }
-      }
-      let participantData = this.participant.participantData.find(participantData => participantData.dataId === relative.dataId);
-      if (participantData != null && participantData.data != null && participantData.data[fieldSetting.columnName] != null) {
-        return participantData.data[fieldSetting.columnName];
+      } else if (personsParticipantData && personsParticipantData.data) {
+        if (personsParticipantData.data[fieldSetting.columnName]) {
+          return personsParticipantData.data[fieldSetting.columnName];
+        } else if (fieldSetting.columnName === 'AGE' && personsParticipantData.data['DATSTAT_DATEOFBIRTH']) {
```

Comment:
could we put that into an action info? so that if another study would like the same calculated but the fields are called differently it would still work?

Reply:
[{"name":"AGE","value":"DATSTAT_DATEOFBIRTH","type":"CALC"}]
so name would point to the columnName and value to dateofbirth and type that it is a CALCulation
not sure where to put the info about how/what to do with the value... maybe type2 could have info about what kind of calculation…

Reply:
I could imagine calculations like 
value: treatment start date
calculate how long participant took a medication
or how long does participant have cancer before started to take medication